components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/LegacyJobOrchestrator.cs

```diff
@@ -775,18 +776,43 @@ IJsonSerializer jsonSerializer
         }
 
         /// <inheritdoc/>
-        public async Task<PublishedNodesEntryModel> GetConfiguredNodesOnEndpointAsync(
+        public async Task<List<OpcNodeOnEndpointModel>> GetConfiguredNodesOnEndpointAsync(
             PublishedNodesEntryModel request,
             CancellationToken ct = default) {
 
             _logger.Information("{nameof} method triggered", nameof(GetConfiguredNodesOnEndpointAsync));
+            var sw = Stopwatch.StartNew();
             await _lockConfig.WaitAsync(ct).ConfigureAwait(false);
+            List<OpcNodeOnEndpointModel> response = new List<OpcNodeOnEndpointModel>();
             try {
-                throw new MethodCallStatusException((int)HttpStatusCode.NotImplemented, "Not Implemented");
+                var nodeFound = false;
+                if (!_publishedNodesEntries.Any()) {
+                    throw new MethodCallStatusException((int)HttpStatusCode.NotFound, "Node not found in endpoint.");
+                }
+                else {
+                    foreach (var entry in _publishedNodesEntries) {
+                        if (request.EndpointUrl == entry.EndpointUrl) {
```

Comment:
We need to have more complete comparison for determining correct endpoint similar to what we have in publish and unpublish nodes, which use `IsSameDataSet(entry, request)` (which internally also checks for `entry.HasSameGroup(request)`).